packages/patterns/src/components/category-selector.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useMemo, useState } from '@wordpress/element';
+import { FormTokenField } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useDebounce } from '@wordpress/compose';
+import { decodeEntities } from '@wordpress/html-entities';
+
+const unescapeString = ( arg ) => {
+	return decodeEntities( arg );
+};
+
+const unescapeTerm = ( term ) => {
+	return {
+		...term,
+		name: unescapeString( term.name ),
+	};
+};
+
+const EMPTY_ARRAY = [];
+const MAX_TERMS_SUGGESTIONS = 20;
+const DEFAULT_QUERY = {
+	per_page: MAX_TERMS_SUGGESTIONS,
+	_fields: 'id,name',
+	context: 'view',
+};
+const slug = 'wp_pattern_category';
+
+export default function CategorySelector( { onCategorySelection } ) {
+	const [ values, setValues ] = useState( [] );
+	const [ search, setSearch ] = useState( '' );
+	const debouncedSearch = useDebounce( setSearch, 500 );
+
+	const { searchResults } = useSelect(
+		( select ) => {
+			const { getEntityRecords } = select( coreStore );
+
+			return {
+				searchResults: !! search
+					? getEntityRecords( 'taxonomy', slug, {
+							...DEFAULT_QUERY,
+							search,
+					  } )
+					: EMPTY_ARRAY,
+			};
+		},
+		[ search ]
+	);
+
+	const suggestions = useMemo( () => {
+		return ( searchResults ?? [] ).map( ( term ) =>
+			unescapeString( term.name )
+		);
+	}, [ searchResults ] );
+
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	async function findOrCreateTerm( term ) {
+		try {
+			const newTerm = await saveEntityRecord( 'taxonomy', slug, term, {
+				throwOnError: true,
+			} );
+			return unescapeTerm( newTerm );
+		} catch ( error ) {
+			if ( error.code !== 'term_exists' ) {
+				throw error;
+			}
+
+			return {
+				id: error.data.term_id,
+				name: term.name,
+			};
+		}
+	}
+
+	function onChange( termNames ) {
+		const uniqueTerms = termNames.reduce( ( acc, name ) => {
+			if (
+				! acc.some( ( n ) => n.toLowerCase() === name.toLowerCase() )
```

Comment:
Nit: Might make grokking this a little smoother if we avoid the abstract variable name. Similar [feedback](https://github.com/WordPress/gutenberg/pull/51078#discussion_r1236233036) was left on the original patterns work.

Reply:
I fixed this [over on the parent PR](https://github.com/WordPress/gutenberg/pull/53835/commits/0b7df8db9086b5ce4f60b38013e0a8f3753663bb) - file was only appearing here due top a missed rebase.

Reply:
No worries, when I reviewed this it was the latest on both PRs, at least on GitHub. Figured we only needed the nit on one.